### ISSUE TITLE
MongoDB auth related issue in MONGO_URI

### ISSUE THREAD

**Issue by smeng9** [pyeve/eve#1478]
### Actual Behavior

MONGO_URI's username, password, and authSource in uri are not parsed correctly

```python
https://github.com/pyeve/eve/blob/master/eve/io/mongo/flask_pymongo.py#L76
```

It uses dbname as authSource, but the authSource should be parsed from options https://pymongo.readthedocs.io/en/stable/api/pymongo/uri_parser.html#pymongo.uri_parser.parse_uri

See URI format https://www.mongodb.com/docs/manual/reference/connection-string/

### Environment

* Python version: 3.10
* Eve version: 2.0

I am willing to open a PR to resolve the issue. Thanks

**Comment by nicolaiarocci** [comment]
hi @smeng9 I'll be happy to review your PR

**Comment by rafrafek** [comment]
How can I stop using URI when using Atlas free? I need srv in my URI, where should I put "srv" without URI?

**Comment by rafrafek** [comment]
I wasted couple of hours today trying to learn Eve, I don't understand why Eve changes `authSource` from default `admin` to `eve`. I had to run debugger and check step-by-step what is the difference between connecting with bare pymongo and connecting with Eve. There is only one difference: `source` set to `eve`. I don't even know how it is being set. There is no place in my code where `eve` could be typed. There is no authSource="eve" in Eve sources. Why Eve changes connection parameters so it cannot authenticate and there is no documentation that even setting authSource in URI does not work.

It works perfectly when using pymongo:
```
mongodb+srv://mySecretName:mySecretPassword@cluster.abcd123.mongodb.net/?retryWrites=true&w=majority&authMechanism=DEFAULT&authSource=admin
```

But Eve magically changes it under the hood to:
```
mongodb+srv://mySecretName:mySecretPassword@cluster.abcd123.mongodb.net/?retryWrites=true&w=majority&authMechanism=DEFAULT&authSource=eve
```

**Comment by micheloe** [comment]
Hi,

I don’t think this is at all possible without using the URI method and in
your case will require a fix in the current code.

Honestly I don’t understand why, if one is using the URI method, the code
is trying to read info from the variables other than DBNAME, which is the
only one that cannot be provided via the URI. I.e. if someone wants / needs
to use the URI method then require them to specify all the options via the
URI, apart from the DBNAME. This has always been confusing me from day one.

Otherwise just let the code only read the MONGO_* settings if they are
absent in the URI and only then expand the connection info kwargs with
these settings.

This will not be backwards compatible but then again so isn’t the current
state of 2.x.

With kind regards,
Michel

Op vr 2 sep. 2022 om 22:18 schreef rafrafek ***@***.***>

**Comment by rafrafek** [comment]
Why can't we just inject MongoClient to Eve and let it handle all connection possibilities? 

It would be so nice to have in `settings.py`:
```python
from pymongo import MongoClient

MONGO_CLIENT = MongoClient("mongodb+srv://...")
```

**Comment by nicolaiarocci** [comment]
Hi all, I just merged #1482. Can you please check current master and confirm if it fixes the problem for you?

**Comment by rafrafek** [comment]
Hello Nicola,

Yes, it fixes the issue immediately. Now I can specify `authSource` and it is being recognized. The second amazing fact is, I don't need to specify `authSource` and it is set to `admin`. I think it is taken from DNS thanks to `srv`.

It looks like this old issue is resolved. Thanks!

Can somebody paste link to this solution here in discussion from 2018: https://groups.google.com/g/python-eve/c/9eJjCSy7wy0 ?

**Comment by micheloe** [comment]
FWIW,

1.1.5 should not display this issue so maybe resorting to this version is
an option for you rafrafek?

Regards,
Michel

I wasted couple of hours today trying to learn Eve, I don't understand why

**Comment by rafrafek** [comment]
@micheloe I don't know if Eve supported `srv` or authSource taken from URI in the past, but I think the issue is now resolved thanks to #1482 and there is no need to downgrade to 1.1.5.
